src/taskgraph/util/vcs.py

```diff
@@ -143,6 +148,15 @@ def working_directory_clean(self, untracked=False, ignored=False):
     def update(self, ref):
         return self.run("update", "--check", ref)
 
+    def find_latest_common_revision(self, base_ref):
+        return self.run(
+            "log",
+            "-r",
+            f"last(ancestors('{base_ref}') and ancestors('{self.head_rev}'))",
```

Comment:
I'm confused, isn't `base_ref` always an ancestor of `head_rev`? In what cases would this not simply return `base_ref`?

Reply:
I haven't tried the `hg` case yet. Maybe it's much simpler to deal with than git. I left an example of such a corner case in this comment.